action.yaml

```diff
@@ -6,20 +6,19 @@ inputs:
   project-dir:
     description: 'Project folder to inspect'
     required: false
-    default: '/github/workspace'
+    default: "${{ github.workspace }}"
   results-dir:
     description: 'Save results to folder'
     required: false
-    default: 'qodana'
-  report-dir:
-    description: 'Save html report to folder'
+    default: "${{ github.workspace }}/qodana"
+  cache-dir:
+    description: 'Save cache to folder'
     required: false
-    default: 'report'
+    default: "/home/runner/work/qodana-cache"
 runs:
-  using: 'docker'
-  image: 'docker://jetbrains/qodana:2020.3-eap'
-  args:
-    - --project-dir=${{inputs.project-dir}}
-    - --results-dir=${{inputs.project-dir}}/${{inputs.results-dir}}
-    - --report-dir=${{inputs.project-dir}}/${{inputs.report-dir}}
-    - --save-report
\ No newline at end of file
+  using: 'composite'
+  steps:
+    - run: docker run -v ${{inputs.project-dir}}:/data/project -v ${{inputs.results-dir}}:/data/results -v  ${{inputs.cache-dir}}:/data/cache jetbrains/qodana:2021.1-eap --save-report
+      shell: bash
+    - run: sudo find ${{inputs.cache-dir}} -type d -exec chmod 755 {} + ; sudo find ${{inputs.cache-dir}} -type f -exec chmod 544 {} +
```

Comment:
Let's run container as `user` instead of fixing permissions afterwards:
```
  steps:
    # create dirs as user, or docker would create as root
    - shell: bash
      run: mkdir -p "${{inputs.results-dir}}" "${{inputs.cache-dir}}"

    - shell: bash
      run: docker run -u $UID -v ${{inputs.project-dir}}:/data/project -v ${{inputs.results-dir}}:/data/results -v ${{inputs.cache-dir}}:/data/cache jetbrains/qodana:2021.1-eap --save-report
```